### PR TITLE
Improved Info.plist merge handling

### LINKED ISSUE #378: Conflicting Info.plist merge strategies
There's a problem when merging Info.plists where the expected result is different depending on which type of key that is merged.

**Case 1 - merging array of dictionaries with SKAdNetworkItems**

PLIST 1
```xml
    <plist version="1.0">
        <dict>
            <key>SKAdNetworkItems</key>
            <array>
                <dict>
                    <key>SKAdNetworkIdentifier</key>
                    <string>7UG5ZH24HU.skadnetwork</string>
                </dict>
            </array>
        </dict>
    </plist>
```

PLIST 2

```xml
    <plist version="1.0">
        <dict>
            <key>SKAdNetworkItems</key>
            <array>
                <dict>
                    <key>SKAdNetworkIdentifier</key>
                    <string>ECPZ2SRF59.skadnetwork</string>
                </dict>
            </array>
        </dict>
    </plist>
```

The expected merge of the SKAdNetworkItems is one array with the two SKAdNetworkIdentifier fields:

EXPECTED
```xml
    <plist version="1.0">
        <dict>
            <key>SKAdNetworkItems</key>
            <array>
                <dict>
                    <key>SKAdNetworkIdentifier</key>
                    <string>ECPZ2SRF59.skadnetwork</string>
                </dict>
                <dict>
                    <key>SKAdNetworkIdentifier</key>
                    <string>7UG5ZH24HU.skadnetwork</string>
                </dict>
            </array>
        </dict>
    </plist>
```

**Case 2 - merging array of dicts with CFBundleURLTypes**

PLIST 1

```xml
    <key>CFBundleURLTypes</key>
    <array>
        <dict>
            <key>CFBundleURLSchemes</key>
            <array>
                <string>fb7886788786688</string>
            </array>
        </dict>
    </array>
```

PLIST 2
```xml
    <key>CFBundleURLTypes</key>
    <array>
        <dict>
            <key>CFBundleTypeRole</key>
            <string>Editor</string>

            <key>CFBundleURLName</key>
            <string>REVERSED_CLIENT_ID</string>
            
            <key>CFBundleURLSchemes</key>
            <array>
                <string>com.fjdkdknvhgjfd</string>
            </array>
        </dict>
    </array>
```

In this case we expect ONE new merged entry:

EXPECTED

```xml
    <key>CFBundleURLTypes</key>
    <array>
        <dict>
            <key>CFBundleTypeRole</key>
            <string>Editor</string>

            <key>CFBundleURLName</key>
            <string>REVERSED_CLIENT_ID</string>
            
            <key>CFBundleURLSchemes</key>
            <array>
                <string>com.fjdkdknvhgjfd</string>
                <string>fb7886788786688</string>
            </array>
        </dict>
    </array>
```

The current code handles the second case, not the first one. How are we going to deal with this? Should we have a list of keys which should be merged according to the first case, and default to a "deep merge" as in case 2?

### PR DESCRIPTION
**Pull request by britzl**
This change improves how `<array>` entries are merged. An array key can now specify a `keep` merge strategy, to keep existing values as-is and add new values to the array. If the array has the `merge` merge strategy (the default), any values in the array will be merged (more specifically `<dict>` values).

Example with the default `merge` strategy:

```xml
A:
<key>My Array</key>
<array>
	<dict>
		<key>Foobar</key>
		<array>
			<string>a</string>
		</array>
	</dict>
</array>

B:
<key>My Array</key>
<array>
	<dict>
		<key>Foobar</key>
		<array>
			<string>b</string>
		</array>
	</dict>
</array>

MERGED:
<key>My Array</key>
<array>
	<dict>
		<key>Foobar</key>
		<array>
			<string>a</string>
			<string>b</string>
		</array>
	</dict>
</array>
```

Same example with the `keep` merge strategy:

```xml
A:
<key merge='keep'>My Array</key>
<array>
	<dict>
		<key>Foobar</key>
		<array>
			<string>a</string>
		</array>
	</dict>
</array>

B:
<key>My Array</key>
<array>
	<dict>
		<key>Foobar</key>
		<array>
			<string>b</string>
		</array>
	</dict>
</array>

MERGED:
<key>My Array</key>
<array>
	<dict>
		<key>Foobar</key>
		<array>
			<string>a</string>
		</array>
	</dict>
	<dict>
		<key>Foobar</key>
		<array>
			<string>b</string>
		</array>
	</dict>
</array>
```

Fixes #378

### CODE REVIEW

File: server/manifestmergetool/src/main/java/com/defold/manifestmergetool/MergePolicy.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.defold.manifestmergetool;
+
+public enum MergePolicy {
+    KEEP,
+    MERGE,
+    REPLACE;
```
Comment (britzl) [inline server/manifestmergetool/src/main/java/com/defold/manifestmergetool/MergePolicy.java:6]: I added REPLACE as a new policy. It is not really needed for this PR but it makes sense to have it.

**Review by AGulev — APPROVED** [review]
Looks good, but 
do we have info about it somewhere in our manuals? I didn't even know we have it for plists (and html as well?)

**Review by ekharkunov — APPROVED** [review]
👍

**Review by ekharkunov — APPROVED** [review]

**Comment by britzl** [comment]
> We do have some documentation here: https://defold.com/manuals/extensions-manifest-merge-tool/
> 
> I'll make sure to update the docs.

Updated the docs: https://github.com/defold/doc/commit/b3e6faa830faceea66c9de29c489b5a9cdb19a62

**Pull request merged**